src/metabase/query_processor/pivot.clj

```diff
@@ -15,16 +15,23 @@
   [breakouts]
   (powerset breakouts))
 
+(defn add-grouping-field
+  "Add the grouping field and expression to the query"
+  [query breakout]
+  (-> query
+      (assoc-in [:query :breakout] breakout)
+      ;;TODO: `pivot-grouping` is not "magic" enough to mark it as an internal thing
+      (update-in [:query :fields]
+                 #(conj % [:expression "pivot-grouping"]))
```

Comment:
I don't think you need a lambda here, update passes the rest of the args along

Reply:
it's the ordering that's the issue if I don't use a lambda:

`(conj [:foo "bar"] nil)` returns `[:foo "bar" nil]` and I don't want to add a nil if there aren't any `:fields`

Reply:
`mbql.u/update-in-unless-empty` or `m/update-in-when`